### PR TITLE
add oneoff script for incrementally migrating users to multi auth

### PR DESCRIPTION
**Pull request by Hamms**
I plan to run this against production data a slice or two at a time, to verify that users migrate as expected, before running it against all remaining unmigrated users.

### CODE REVIEW

File: bin/oneoff/migrate_users_to_multi_auth.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+
+SLICE_SIZE = 1_000
+
+def migrate_batches(limit=nil)
+  limit = limit.to_i if limit
+  unmigrated_users = User.where.not(provider: User::PROVIDER_MIGRATED)
+  unmigrated_users_count = 40_000_000 # estimate
```
Comment (Hamms) [inline bin/oneoff/migrate_users_to_multi_auth.rb:10]: estimated this via an SQL select on the reporting DB.

File: bin/oneoff/migrate_users_to_multi_auth.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+
+SLICE_SIZE = 1_000
+
+def migrate_batches(limit=nil)
+  limit = limit.to_i if limit
+  unmigrated_users = User.where.not(provider: User::PROVIDER_MIGRATED)
```
Comment (islemaster) [inline bin/oneoff/migrate_users_to_multi_auth.rb:9]: Do we need to `with_deleted` include soft-deleted users, or are we officially leaving those behind?

**Review by islemaster — APPROVED** [review]
Woohoo!

File: bin/oneoff/migrate_users_to_multi_auth.rb (hunk 2 above)
Comment (Hamms) [reply bin/oneoff/migrate_users_to_multi_auth.rb:9]: My plan is to leave them behind, then wait 30 days after this script finishes to begin tearing out support for nonmigrated users to ensure that they are fully gone.

**Pull request merged**